string.c

```diff
@@ -11307,6 +11307,21 @@ rb_str_delete_suffix(VALUE str, VALUE suffix)
     return rb_str_subseq(str, 0, RSTRING_LEN(str) - suffixlen);
 }
 
+/*
+ *  call-seq:
+ *    ensure_suffix(suffix) -> self or new_string
+ *
+ *  :include: doc/string/ensure_suffix.rdoc
+ *
+ */
+static VALUE
+rb_str_ensure_suffix(VALUE str, VALUE suffix)
+{
+    if (rb_str_end_with(1, &suffix, str)) return str;
+
+    return rb_str_plus(str, suffix);
```

Comment:
I've used `+` here because I don't know if it is possible or desirable to use string interpolation in this case.